model/Signal.py

```diff
@@ -78,11 +116,15 @@ def iasd(self, x):                                     # integrated absolute of
     def iatd(self, x):                                     # integrated absolute of third derivative
         return (sum(abs(self.derivative(self.derivative(self.derivative(x))))))
     
-    def sliding_avg(self, x, w=1):                         # sliding average with window size w
-        return np.convolve(x, np.ones(w), 'valid') / w
+    def sliding_avg(self, x, w=1): 
```

Comment:
Again try not to modify other functions. This is a relatively small codebase but you don't know where else this function is being used and that is very likely to break things. If you want this functionality, implement it outside of this function. And convolve is a vector operation anyway I think. I'm not too sure about that though. But don't modify already existing functions that are not a part of your commit.